### DISCUSSION TITLE
How to compress the data while we send data over kafka bridge from HTTP request .

### DISCUSSION THREAD

**Discussion by suresh-tattikota** [strimzi/strimzi-kafka-bridge#893] [Q&A]
How to compress the data while we send data over kafka bridge from HTTP request .
how to compress request body (gzip/snappy) while doing POST call to Kafka bridge over HTTP request

**Comment by scholzj** [comment]
I assume you are talking about the compression between the HTTP client and the bridge? I think the only option to compress the message body is to compress it yourself and send it as binary payload (base64 encoded). But keep in mind that it will be send this way to Kafka as well.

**Reply by ppatierno** [reply]
Yes Jakub is right. The HTTP server part of the bridge doesn't expose any way to configure the compression. It means the body itself cannot be compressed following the usual standard algorithm but you can do what Jakub suggested, which is of course different but the only viable way at the moment.

**Reply by suresh-tattikota** [reply]
Thank you for you reply Jakub , paolo 
i have one more question regarding the same 
do we have any authentication and authorization available from client http to strimzi kafka bridge?

**Reply by ppatierno** [reply]
No I am sorry. Currently, the only viable way it to put an API gateway in front of the bridge to get AuthN/AuthZ for HTTP clients. It's on our roadmap to add such a support but there is really not plan or ETA for it.

**Reply by suresh-tattikota** [reply]
thank you paolo
